astroquery/nasa_exoplanet_archive/core.py

```diff
@@ -0,0 +1,579 @@
+
+                else:  # pragma: nocover
+                    warnings.warn("Unrecognized unit: '{0}'".format(unit), AstropyWarning)
+
+            # Here we're figuring out out if the column is masked because this doesn't
+            # play nice with quantities so we need to keep track of the mask separately.
+            try:
+                column_masks[col] = data[col].mask
+            except AttributeError:
+                pass
+            else:
+                data[col].mask[:] = False
+
+            # Deal with strings consistently
+            if data[col].dtype == np.object:
+                data[col] = data[col].astype(str)
+
+            data[col].unit = unit
+            column_data.append(data[col])
+
+        # Build the new `QTable` and copy over the data masks if there are any
+        result = QTable(column_data, names=column_names, masked=len(column_masks) > 0)
+        for key, mask in column_masks.items():
+            result[key].mask = mask
+
+        return result
+
+    def _parse_result(self, response, verbose=False):
+        """
+        Parse the result of a `~requests.Response` object and return an `~astropy.table.Table`
+
+        Parameters
+        ----------
+        response : `~requests.Response`
+            The response from the server.
+        verbose : bool
+            Currently has no effect.
+
+        Returns
+        -------
+        data : `~astropy.table.Table` or `~astropy.table.QTable`
+        """
+
+        # Extract the decoded body of the response
+        text = response.text
+
+        # Raise an exception if anything went wrong
+        self._handle_error(text)
+
+        # Parse the requested format to figure out how to parse the returned data
+        fmt = response.requested_format.lower()
+        if "ascii" in fmt or "ipac" in fmt:
+            data = ascii.read(text, format="ipac", fast_reader=False, converters=CONVERTERS)
+        elif "csv" in fmt:
+            data = ascii.read(text, format="csv", fast_reader=False, converters=CONVERTERS)
+        elif "bar" in fmt or "pipe" in fmt:
+            data = ascii.read(text, fast_reader=False, delimiter="|", converters=CONVERTERS)
+        elif "xml" in fmt or "table" in fmt:
+            data = parse_single_table(io.BytesIO(response.content)).to_table()
+        else:
+            data = ascii.read(text, fast_reader=False, converters=CONVERTERS)
+
+        # Fix any undefined units
+        data = self._fix_units(data)
+
+        # For backwards compatibility, add a `sky_coord` column with the coordinates of the object
+        # if possible
+        if "ra" in data.columns and "dec" in data.columns:
+            data["sky_coord"] = SkyCoord(
+                ra=u.Quantity(data["ra"], unit=u.deg), dec=u.Quantity(data["dec"], unit=u.deg)
+            )
+
+        if not data:
+            warnings.warn("Query returned no results.", NoResultsWarning)
+
+        return data
+
+    def _handle_deprecated_arguments(self, **kwargs):
```

Comment:
`astropy.utils.decorators` have a nice decorator to do this `deprecated_renamed_argument`, please use that one instead.
(note, we bumped the minimum astropy version to 3.1, so I believe it should work, for prior versions we shipped a bundled copy in `astroquery.utils.decorators`)

Reply:
Done.